src/misc/filters.js

```diff
@@ -321,3 +321,108 @@ ngeo.misc.filters.trustHtmlFilter = function($sce) {
 };
 
 ngeo.misc.filters.filter('ngeoTrustHtml', ngeo.misc.filters.trustHtmlFilter);
+
+
+/**
+ * A filter used to format a time duration in seconds into a more
+ * readable form.
+ * Only the two largest units will be shown.
+ *
+ * Examples:
+ *      {{42 | ngeoDuration}} => 42 seconds
+ *      {{132 | ngeoDuration}} => 2 minutes 12 seconds
+ *      {{3910 | ngeoDuration}} => 1 hour 5 minutes
+ *        -> Note: the remaining 10 seconds will be dropped
+ *
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
+ * @return {ngeox.duration} Function used to format a time duration
+ *  in seconds into a string.
+ * @ngInject
+ * @ngdoc filter
+ * @ngname ngeoDuration
+ */
+ngeo.misc.filters.Duration = function(gettextCatalog) {
+  // time unit enum
+  const TimeUnits = Object.freeze({
+    SECONDS: Symbol('seconds'),
+    MINUTES: Symbol('minutes'),
+    HOURS: Symbol('hours'),
+    DAYS: Symbol('days')
+  });
+
+  /**
+   * @param {number} amount Amount of time.
+   * @param {symbol} unit Unit of time.
+   * @return {string} formatted and translated string
+   */
+  const pluralize = function(amount, unit) {
```

Comment:
angular-gettext already have something to manage plurials, it's not working?

Reply:
yes, see bfc751a185488fed9362bf82b844e82b00cc575a